src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -306,8 +306,9 @@ def __init__(self, config: LlavaNextConfig):
         self.vision_tower = AutoModel.from_config(config.vision_config)
 
         self.multi_modal_projector = LlavaNextMultiModalProjector(config)
-
-        self.image_newline = nn.Parameter(torch.empty(config.text_config.hidden_size, dtype=self.dtype))
+        self.image_newline = nn.Parameter(
+            torch.randn(config.text_config.hidden_size, dtype=config.torch_dtype or self.dtype)
```

Comment:
We needed some way to set this half precision in autocast settings. Let me know if this is incorrect. I am still unclear when to use `torch_dtype` and when not to.

Reply:
Also I should note I am not quite certain `randn` is correct here but the `empty` was causing NaNs.

Reply:
Just from looking at this, I suspect the issue is that the param `self.image_newline` is being create as an empty tensor, but it's not then being initialized in `_init_weights`

Reply:
Right I will move this initialization to _init_weights

Reply:
At a second glance it looks sufficient to do it on initialization. The original repo just had a a separate function to initialize the vision tower.